nexus/db-queries/src/db/datastore/mod.rs

```diff
@@ -200,29 +200,53 @@ impl DataStore {
             .unwrap();
     }
 
-    // TODO-security This should be deprecated in favor of pool_authorized(),
-    // which gives us the chance to do a minimal security check before hitting
-    // the database.  Eventually, this function should only be used for doing
-    // authentication in the first place (since we can't do an authz check in
-    // that case).
-    fn pool(&self) -> &bb8::Pool<ConnectionManager<DbConnection>> {
-        self.pool.pool()
-    }
-
-    pub(super) async fn pool_authorized(
+    async fn pool_authorized(
         &self,
         opctx: &OpContext,
     ) -> Result<&bb8::Pool<ConnectionManager<DbConnection>>, Error> {
         opctx.authorize(authz::Action::Query, &authz::DATABASE).await?;
         Ok(self.pool.pool())
     }
 
+    /// Returns a connection to a connection from the database connection pool.
+    pub(super) async fn pool_connection_authorized(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<bb8::PooledConnection<ConnectionManager<DbConnection>>, Error>
+    {
+        let pool = self.pool_authorized(opctx).await?;
+        let connection = pool.get().await.map_err(|err| {
+            Error::internal_error(&format!(
+                "Failed to access DB connection: {err}"
+            ))
```

Comment:
This is probably not an internal error, but rather a 503.  I think we had logic before in db/error.rs to figure this out?

Reply:
Good catch, I think the populator tests caught this too. Updated.